modules/planning/common/trajectory/trajectory_stitcher.cc

```diff
@@ -107,20 +107,24 @@ std::vector<TrajectoryPoint> TrajectoryStitcher::ComputeStitchingTrajectory(
     return ComputeReinitStitchingTrajectory(vehicle_state);
   }
 
-  double forward_rel_time = veh_rel_time + planning_cycle_time;
+  double forward_rel_time =
+      prev_trajectory->TrajectoryPointAt(matched_index).relative_time() +
+      planning_cycle_time;
+
   std::size_t forward_index =
       prev_trajectory->QueryNearestPoint(forward_rel_time);
 
+  ADEBUG << "matched_index: " << matched_index;
   std::vector<TrajectoryPoint> stitching_trajectory(
-      prev_trajectory->trajectory_points().begin() + matched_index,
+      prev_trajectory->trajectory_points().begin() +
+          std::max(0, static_cast<int>(matched_index - 1)),
       prev_trajectory->trajectory_points().begin() + forward_index + 1);
 
-  const double zero_time = veh_rel_time;
-  const double zero_s =
-      prev_trajectory->TrajectoryPointAt(forward_index).path_point().s();
+  const double zero_s = matched_point.path_point().s();
 
   for (auto& tp : stitching_trajectory) {
-    tp.set_relative_time(tp.relative_time() - zero_time);
+    tp.set_relative_time(tp.relative_time() + prev_trajectory->header_time() -
+                         current_timestamp);
```

Comment:
shall we use tp.relative_time() - prev_trajectory->TrajectoryPointAt(matched_index).relative_time(); ?